pkg/chain/common/gen/Makefile

```diff
@@ -0,0 +1,45 @@
+# If contracts_dir is not defined by a Makefile using this common Makefile default
+# it to a `contracts/` subdirectory under solidity_dir.
+ifndef contracts_dir
+	contracts_dir = $(addsuffix /contracts/,${solidity_dir})
+endif
+
+# Solidity files prefixed with the solidity directory path.
+# Multiple patterns have to be defined to get the files recursively.
+solidity_files := $(wildcard ${contracts_dir}/*.sol) $(wildcard ${contracts_dir}/*/*.sol)
+
+# Go bindings generated for the solidity contracts.
+contract_files = $(addprefix contract/,$(addsuffix .go,${required_contracts}))
+
+# Function to run solidity sources comilation.
+# Expects two arguments:
+# 1 - dependencies mapping for contracts import
+# 2 - solidity source file path
+solc_generate = solc $(1) \
+		--allow-paths ${solidity_dir} \
+		--overwrite \
+		--abi \
+		-o abi $(2)
+
+all: gen_contract_go
```

Comment:
When I run `make` or `make all` from `pkg/chain` I get:
`make: *** No rule to make target "all".  Stop.`

When I run `make` or `make all` from a given dir like `pkg/chain/random-beacon/gen` I get: 
```
solc  @openzeppelin/=/node_modules/@openzeppelin/ @keep-network/=/node_modules/@keep-network/ @threshold-network/=/node_modules/@threshold-network/ @thesis/=/node_modules/@thesis/ --allow-paths  --overwrite --abi -o abi  
No input files given. If you wish to use the standard input please specify "-" explicitly.
```

Do you have similar errors? Is `make` command supported?

Reply:
You need to specify `SOLIDITY_DIR` env variable.
Like here:
https://github.com/keep-network/keep-core/blob/f60b46da4018840e4ff3ee63a3bd5e3e59837395/pkg/chain/random-beacon/gen/gen.go#L3

Reply:
Please try again after the latest changes I pushed.

Reply:
Now it works.